core/sql/executor/ExExeUtilGet.cpp

```diff
@@ -5795,40 +5833,51 @@ short ExExeUtilHiveMDaccessTcb::work()
             str_cpy(infoCol->colName, 
                     (hcd ? hcd->name_ : hpd->name_), 256, ' ');
 
-            infoCol->fsDatatype = 
-              getFSTypeFromHiveColType(hcd ? hcd->type_ : hpd->type_);
-
-	    if (infoCol->fsDatatype < 0)
+            Lng32 fstype = -1;
+            Lng32 length = -1;
+            Lng32 precision = -1;
+            Lng32 scale = -1;
+            char sqlType[50];
+            char displayType[100];
+            char charset[50];
+            retcode = 
+              getTypeAttrsFromHiveColType(hcd ? hcd->type_ : hpd->type_,
+                                          htd->getSDs()->isOrcFile(),
+                                          fstype, length, precision, scale,
+                                          sqlType, displayType, charset);
+                 
+	    if (retcode < 0)
 	      {
+                // add a warning and continue.
 		char strP[300];
-		sprintf(strP, "Datatype %s is not supported.", 
-                        (hcd ? hcd->type_ : hpd->type_));
-		*diags << DgSqlCode(-CLI_GET_METADATA_INFO_ERROR)
+		sprintf(strP, "Datatype %s for column '%s' in table %s.%s.%s is not supported. This table will be ignored.", 
```

Comment:
300 is not long enough. The column name, schema name and table name can each be up to 256 characters.

Reply:
Why not use snprintf(strP, sizeof(strP), ...)?

Reply:
In #1504 I am changing this type of populating error to use ExRaiseSqlError. This needs to be changed to use ExRaiseSqlError because diags could be null